### PR TITLE
Port Sniffer && Data Structures

### PR DESCRIPTION
**Pull request by assmass13**
Port Sniffer for TCP. UDP. Network task is developed.
Data structures via http-server are developed.

### CODE REVIEW

File: submissions/assmass13/port-sniffer/sniffer.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    exit(0);
+  } else if (
+    args.indexOf('--host') === -1 ||
+    !args[args.indexOf('--host') + 1]
+  ) {
+    stdoutWrite(hostErrorMessage);
+    exit(1);
+  }
+  return args.reduce((result, item, index, arr) => {
+    const newResult = { ...result };
```
Comment (AMashoshyna) [inline submissions/assmass13/port-sniffer/sniffer.js]: Why do you create new object on every iteration?

File: submissions/assmass13/port-sniffer/sniffer.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+    });
+  }).catch(e => {
+    stdoutWrite(lookupErrorMessage(e));
+    exit(1);
+  });
+}
+
+function getPortLimits(portsRange) {
+  const portArr = portsRange.split('-').map(portNum => parseInt(portNum, 10));
+  if (
```
Comment (AMashoshyna) [inline submissions/assmass13/port-sniffer/sniffer.js]: please put this check into separate function

File: submissions/assmass13/port-sniffer/sniffer.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    socket.connect(port, address);
+  });
+}
+
+function getAddressOpenPorts(address, startPort, endPort) {
+  const openPorts = [];
+  for (let port = startPort; port < endPort; port += 1) {
+    openPorts.push(scanAddressPort(address, port));
+  }
+  return Promise.all(openPorts).then(values => values.filter(Number));
```
Comment (AMashoshyna) [inline submissions/assmass13/port-sniffer/sniffer.js]: Is port 0 an option?

**Review by AMashoshyna — COMMENTED** [review]
Good job! Nice code structuring. Please see below some comments.

File: submissions/assmass13/port-sniffer/sniffer.js (hunk 2 above)
Comment (assmass13) [reply submissions/assmass13/port-sniffer/sniffer.js]: fixed

File: submissions/assmass13/port-sniffer/sniffer.js (hunk 3 above)
Comment (assmass13) [reply submissions/assmass13/port-sniffer/sniffer.js]: fixed

File: submissions/assmass13/port-sniffer/sniffer.js (hunk 1 above)
Comment (assmass13) [reply submissions/assmass13/port-sniffer/sniffer.js]: https://eslint.org/docs/2.0.0/rules/no-param-reassign

Comment (AMashoshyna) [reply submissions/assmass13/port-sniffer/sniffer.js]: Oh, I see. Here is how I would tackle this:
```js
  return args.reduce((result, item, index, arr) => {
    if (item.startsWith('--') && arr[index + 1]) {
      return { ...result, [item]: arr[index + 1] };
    }
    return result;
  }, {});
```
Just a suggestion.

Comment (assmass13) [reply submissions/assmass13/port-sniffer/sniffer.js]: fixed

**Pull request closed without merging**